### PR TITLE
Skip the specified classes when breaking on exception or stepping

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

### CODE REVIEW

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StepRequestHandler.java [hunk 1]
```diff
@@ -133,21 +134,24 @@ private void handleDebugEvent(DebugEvent debugEvent, IDebugSession debugSession,
...
-                        // Check if the step into operation stepped through the filtered code and stopped at an un-filtered location.
-                        if (threadState.stackDepth + 1 < thread.frameCount()) {
-                            // Create another stepOut request to return back where we started the step into.
-                            threadState.pendingStepRequest = DebugUtility.createStepOutRequest(thread,
-                                    context.getStepFilters().classNameFilters);
-                            threadState.pendingStepRequest.enable();
-                            debugEvent.shouldResume = true;
-                            return;
-                        }
+                        // // Check if the step into operation stepped through the filtered code and stopped at an un-filtered location.
```
Comment (Eskibear) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/StepRequestHandler.java:137]: unused code?

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaClassFilter.java [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+                    node = parent.children.get(names[i]);
+                } else {
+                    node = new TrieNode(names[i]);
+                    parent.children.put(names[i], node);
+                }
+
+                parent = node;
+            }
+
+            parent.isLeaf = true;
```
Comment (jdneo) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaClassFilter.java:253]: set a 'parent' node to leaf is a little bit confusing to me. How about rename the `parent` to `currentNode`?

File: com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaClassFilter.java [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        }
+
+        return false;
+    }
+
+    private static boolean isJDKPackageFragmentRoot(IPackageFragmentRoot root) throws JavaModelException {
+        if (root.getRawClasspathEntry() != null) {
+            IPath path = root.getRawClasspathEntry().getPath();
+            return path != null && path.segmentCount() > 0
+                && Objects.equals("org.eclipse.jdt.launching.JRE_CONTAINER", path.segment(0));
```
Comment (jdneo) [inline com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaClassFilter.java:128]: Consider using `JavaRuntime.JRE_CONTAINER`?

Comment (testforstephen) [reply com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/JavaClassFilter.java]: Nice!

**Pull request merged**